encoder_union_test.go

```diff
@@ -550,3 +550,94 @@ func TestEncoder_UnionInterfaceNotInSchema(t *testing.T) {
 
 	assert.Error(t, err)
 }
+
+func TestEncoder_UnionResolver(t *testing.T) {
```

Comment:
I didn't find any tests testing this behavior so I added it, specifically the test case `Go int as Avro long` tests the line added in this PR. It's written in a different style than the other tests in this repo, if you prefer I can split it up into separate tests.